src/crosswalk/plots.py

```diff
@@ -0,0 +1,374 @@
+    pred_df = pd.DataFrame({'obs_method': obs_method,  
+                            'prev': 0.1, 
+                            'prev_se': 0.1}, index=[0])
+    # if it's continuous variable, take median 
+    for var in continuous_variables:
+        pred_df[var] = np.median(cwdata.covs[var])
+    
+    # predict effect
+    y_pred = cwmodel.adjust_orig_vals(
+        df=pred_df, 
+        orig_dorms = "obs_method", 
+        orig_vals_mean = "prev",  
+        orig_vals_se = "prev_se"
+        ).to_numpy()
+    y_pred = np.ravel(y_pred)
+    y_mean, y_sd = y_pred[2], y_pred[3]
+    
+    # Statistics in title 
+    y_lower, y_upper = y_mean - 1.96*y_sd, y_mean + 1.96*y_sd
+    p_value = cw.utils.p_value(np.array([y_mean]), np.array([y_sd]))[0]
+    content_string = f"Mean effect: {np.round(y_mean, 3)}\
+    (95% CI: {np.round(y_lower, 3)} to {np.round(y_upper, 3)});\
+    p-value: {np.round(p_value, 4)}"
+
+    # triangle
+    max_se = cwdata.obs_se.max()
+    se_domain = np.arange(0, max_se*1.1, max_se / 100)
+    se_lower = y_mean - (se_domain*1.96)
+    se_upper = y_mean + (se_domain*1.96)
+    
+    sns.set_style('darkgrid')
+    plt.rcParams['axes.edgecolor'] = '0.15'
+    plt.rcParams['axes.linewidth'] = 0.5
+    plt.figure(figsize=(10,8))
+    plt.fill_betweenx(se_domain, se_lower, se_upper, color='white', alpha=0.75)
+    plt.axvline(y_mean, 0, 1 - (0.025*max(se_domain) / (max(se_domain)*1.025)), 
+                color='black', alpha=0.75, linewidth=0.75)
+    plt.plot(se_lower, se_domain, color='black', linestyle='--', linewidth=0.75)
+    plt.plot(se_upper, se_domain, color='black', linestyle='--', linewidth=0.75)
+    plt.ylim([-0.025*max(se_domain), max(se_domain)])
+    plt.xlabel('Effect size', fontsize=10)
+    plt.xticks(fontsize=10)
+    plt.ylabel('Standard error', fontsize=10)
+    plt.yticks(fontsize=10)
+    plt.axvline(0, color='mediumseagreen', alpha=0.75, linewidth=0.75)
+    # Plot inlier and outlier
+    for key, value in plot_key.items():
+        plt.plot(
+            data_df.loc[data_df.plot_guide == key, 'y'],
+            data_df.loc[data_df.plot_guide == key, 'se'],
+            'o',
+            markersize=5,
+            marker=value[0], markerfacecolor=value[1], markeredgecolor=value[2], 
+            markeredgewidth=0.6, alpha=.6, label=key
+        )
+
+    plt.legend(loc='upper left', frameon=False)
+    plt.gca().invert_yaxis()
+    # Plot title
+    if plot_note is not None:
+        plt.title(content_string, fontsize=10)
+        plt.suptitle(plot_note, y=1.01, fontsize=12)
+    else:
+        plt.title(content_string, fontsize=10)
+    # Save plots
+    if write_file:
+        assert cwdir is not None, "cwdir is not specified!"
+        outfile = os.path.join(cwdir, file_name + '.pdf')
+        plt.savefig(outfile, orientation='landscape', bbox_inches='tight')
+        print(f"Funnel plot saved at {outfile}")
+    else:
+        plt.show()
+    plt.clf()
+
+
+def sizes_to_slices(sizes):
```

Comment:
There are two same functions in `plots.py` and `utils.py`. Will be nice if we only keep this in `utils.py` later on :)

Reply:
On my tests using cw.utils would work; but not for Reed since it is not in the crosswalk package yet

Reply:
I see will straighten it out after installation.